pkg/trie/triedb/triedb.go

```diff
@@ -685,4 +699,166 @@ func (t *TrieDB) lookupNode(hash common.Hash) (StorageHandle, error) {
 	}), nil
 }
 
+func (t *TrieDB) commit() error {
+	log.Printf("Committing trie changes to db")
+	log.Printf("%d nodes to remove from db", len(t.deathRow))
+
+	for hash := range t.deathRow {
+		err := t.db.Del(hash[:])
```

Comment:
could you open a batch, do the modifications and then flush applying all these modifications to the database at once? I would say it is better than hit the db every time (you might want to rollback if something went wrong)

Reply:
Sure! good idea

Reply:
I added support for batches, not only for deletes but also for the inserts / updates. Thanks for the reminder.